src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -287,29 +336,235 @@ public TestResult findCorrespondingResult(String id) {
+             */
+            ChartUtil.generateGraph(req, rsp, createChart(req, buildDataSetPerProject(req)), calcDefaultSize());
+        } else if ((index >= 0 || projectLevel.equals(ALLPROJECTS)) && trendType.equals(LENGTHY_TESTS)) {
+
+            /*
+             * This method generates the trends depicting no. of passed testcases which took longer duration
+             * to run in the given build.
+             */
+            ChartUtil.generateGraph(req, rsp, createChart(req, buildLengthyTestDataset(req)), calcDefaultSize());
+        } else if ((index >= 0 || projectLevel.equals(ALLPROJECTS)) && trendType.equals(FLAKY_TESTS)) {
+
+            /*
+             * This method generates the trends depicting no. of passed and failed testcases which were
+             * inconsistently failing or passing i.e. flappy behaviour.
+             */
+            ChartUtil.generateGraph(req, rsp, createXYChart(req, buildFlapperDataset(req)), calcDefaultSize());
+        } else {
+
+            /*
+             * This method is invoked when a user deliberately fires a wrong url with invalid query
+             * parameters and it depicts trend showing no. of passed, failed and skipped testcases for all
+             * projects.
+             */
+            ChartUtil.generateGraph(req, rsp, createChart(req, buildDataSet(req)), calcDefaultSize());
+        }
     }
 
     /**
      * Generates a clickable map HTML for {@link #doGraph(StaplerRequest, StaplerResponse)}.
      */
-    public void doGraphMap( StaplerRequest req, StaplerResponse rsp) throws IOException {
-        if(req.checkIfModified(run.getTimestamp(),rsp))
+    public void doGraphMap (StaplerRequest req, StaplerResponse rsp) throws IOException {
+        if (req.checkIfModified(run.getTimestamp(), rsp))
             return;
-        ChartUtil.generateClickableMap(req,rsp,createChart(req,buildDataSet(req)),calcDefaultSize());
+
+        /*
+         * The Utility method to generate a mapping from chart coordinates to url to redirect to on
+         * clicking the trend.
+         */
+        doGraphMapUtil(req, rsp);
+    }
+
+    /**
+     * A utility method for constructing a mapping from chart coordinates to the url to redirect to on
+     * clicking the trend.
+     * @param req HTTP request message for the image of trend.
+     * @param rsp HTTP response message for the requested image.
+     * @throws IOException In case an exception occurs in
+     * {@link ChartUtil#generateClickableMap(StaplerRequest, StaplerResponse, JFreeChart, Area)}
+     */
+    public void doGraphMapUtil (StaplerRequest req, StaplerResponse rsp) throws IOException {
+        String projectLevel = getParameter(req, PROJECTLEVEL);
+        String trendType = getParameter(req, TRENDTYPE);
+        int index = Arrays.binarySearch(projectList, projectLevel);
+        if ((index >= 0 || projectLevel.equals(ALLPROJECTS)) && trendType.equals(BUILD_ANALYSIS)) {
```

Comment:
`if ((index >= 0 || projectLevel.equals(ALLPROJECTS)) ` looks common to 3 checks. Can be written as single `if` condition. You can use switch case for `trendType`.

Also, please indent it properly with a line break in b/e condition like 
```
if () {
}
else {
}
```

Reply:
Done.